controller/controller_test.go

```diff
@@ -312,6 +312,70 @@ func TestController_Start(t *testing.T) {
 	mock.AssertExpectationsForObjects(t, fleetMock.Mock)
 }
 
+
+func TestController_Destroy(t *testing.T) {
+	RegisterTestingT(t)
+
+	// Mocks
+	controller, fleetMock := GivenController()
+	fleetMock.On("GetStatusWithMatcher", mock.AnythingOfType("func(string) bool")).Return(
+		[]fleet.UnitStatus{
+			{
+				Name: "test-main@1.service",
+			},
+			{
+				Name: "test-sidekick@1.service",
+			},
+		},
+		nil,
+	).Once()
+	fleetMock.On("Destroy", "test-main@1.service").Return(nil).Once()
+	fleetMock.On("Destroy", "test-sidekick@1.service").Return(nil).Once()
+
+	// Execute test
+	req := Request{
+		Group:    "test",
+		SliceIDs: []string{"1"},
```

Comment:
We also talked about that we maybe want a separate Request struct for the Submit case so the `Unit` field is not always there. See https://github.com/giantswarm/giantswarm/issues/527